parquet/src/test/java/org/apache/iceberg/parquet/TestBloomRowGroupFilter.java

```diff
@@ -0,0 +1,950 @@
+    }
+  }
+
+  @Test
+  public void testStructFieldGt() {
+    for (int i = INT_MIN_VALUE - 20; i < INT_MAX_VALUE + 20; i++) {
+      boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, greaterThan("struct_not_null.int_field", i))
+          .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      Assert.assertTrue("Should read: bloom filter doesn't help", shouldRead);
+    }
+  }
+
+  @Test
+  public void testStructFieldGtEq() {
+    for (int i = INT_MIN_VALUE - 20; i < INT_MAX_VALUE + 20; i++) {
+      boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, greaterThanOrEqual("struct_not_null.int_field", i))
+          .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      Assert.assertTrue("Should read: bloom filter doesn't help", shouldRead);
+    }
+  }
+
+  @Test
+  public void testStructFieldEq() {
+    for (int i = INT_MIN_VALUE - 20; i < INT_MAX_VALUE + 20; i++) {
+      boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, equal("struct_not_null.int_field", i))
+          .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      if (i >= INT_MIN_VALUE && i <= INT_MAX_VALUE) {
+        Assert.assertTrue("Should read: value within range", shouldRead);
+      } else {
+        Assert.assertFalse("Should not read: value outside range", shouldRead);
+      }
+    }
+  }
+
+  @Test
+  public void testStructFieldNotEq() {
+    for (int i = INT_MIN_VALUE - 20; i < INT_MAX_VALUE + 20; i++) {
+      boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, notEqual("struct_not_null.int_field", i))
+          .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+      Assert.assertTrue("Should read: bloom filter doesn't help", shouldRead);
+    }
+  }
+
+  @Test
+  public void testCaseInsensitive() {
+    // the column name is required. If setting caseSentitive to true, ValidationException: Cannot find field 'Required'
+    boolean shouldRead = new ParquetBloomRowGroupFilter(SCHEMA, equal("Required", "Req"), false)
+        .shouldRead(parquetSchema, rowGroupMetadata, bloomStore);
+    Assert.assertFalse("Should skip: contains only 'req'", shouldRead);
+  }
+
+  @Test
+  public void testMissingBloomFilterForColumn() {
+    TestHelpers.assertThrows("Should complain about missing bloom filter",
+        IllegalStateException.class, "Failed to read required bloom filter for id: 10",
+        () -> new ParquetBloomRowGroupFilter(SCHEMA, equal("some_nulls", "some"))
+            .shouldRead(parquetSchema, rowGroupMetadata, new DummyBloomFilterReader(null, rowGroupMetadata)));
+  }
+
+  private static class DummyBloomFilterReader extends BloomFilterReader {
+    DummyBloomFilterReader(ParquetFileReader fileReader, BlockMetaData block) {
+      super(fileReader, block);
+    }
+
+    @Override
+    public BloomFilter readBloomFilter(ColumnChunkMetaData meta) {
+      return null;
+    }
+  }
+
+  @Test
+  public void testIntegerIn() {
+    for (int i = 0; i < INT_VALUE_COUNT; i++) {
```

Comment:
I don't think this needs to be tested in loops like this. I think the case you want to test is that `in` behaves like `or`. So you should have a case where all values are present, no values are present, and where only one value is present.

Reply:
Changed. Thanks!